src/sql-parser/src/parser.rs

```diff
@@ -2920,6 +2927,34 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_alter_secret(&mut self) -> Result<Statement<Raw>, ParserError> {
+        let if_exists = self.parse_if_exists()?;
+        let name = self.parse_object_name()?;
+
+        Ok(match self.expect_one_of_keywords(&[AS, RENAME])? {
+            AS => {
+                let value = self.parse_expr()?;
+                Statement::AlterSecret(AlterSecretStatement {
+                    secret_name: name,
+                    if_exists,
+                    value,
+                })
+            }
+            RENAME => {
+                self.expect_keyword(TO)?;
+                let to_item_name = self.parse_identifier()?;
+
+                Statement::AlterObjectRename(AlterObjectRenameStatement {
+                    object_type: ObjectType::Index,
```

Comment:
Ah, and here's the source of that bug I mentioned below!

Reply:
well spotted!